### PR TITLE
Add rio core package

### PR DESCRIPTION
**Pull request by epicallan**

### CODE REVIEW

File: core/package.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name:       rio-core
```
Comment (mbj) [inline core/package.yaml]: should we name the package `mrio-core`? Its trivial to rename to `rio-core` later on upstreaming.

But we do not clash with upstream RIO meanwhile.

File: core/package.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+synopsis:   Minimal version of rio with the RIO core type.
+homepage:   https://github.com/strainge/rio-core#readme
```
Comment (mbj) [inline core/package.yaml]: This needs to be changed to the new repository name.

File: core/package.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+maintainer: Allan Lukwago <epicallan.al@gmail.com>, Markus Schirp mbj@schirp-ds.com
```
Comment (mbj) [inline core/package.yaml]: `schirp-dso.com`

File: core/package.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+copyright:  2020 Strainge Inc.
+license:    MIT
+github:     strainge/rio-core
```
Comment (mbj) [inline core/package.yaml]: `strainge/mrio`.

File: core/package.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+version:    0.0.1
+
+dependencies:
+- base >= 4.10 && < 10
```
Comment (mbj) [inline core/package.yaml:11]: Declaring we are compatible with any base upto excluding `10` seems to be very wide, is this a good default?

File: core/src/RIO/Core.hs [hunk 6]
```diff
@@ -0,0 +1,70 @@
+-- | This module takes code from RIO.Prelude.RIO and reproduces it here
+--  so that we can propose rio-core package to the rio team.
+--
+{-# OPTIONS -Wno-noncanonical-monoid-instances #-}
+module RIO.Core
```
Comment (mbj) [inline core/src/RIO/Core.hs]: Till we upstream, should we name it `MRIO.Core` ?

File: core/package.yaml (hunk 1 above)
Comment (epicallan) [reply core/package.yaml]: Yes I agree. Lets do so.

File: core/package.yaml (hunk 5 above)
Comment (epicallan) [reply core/package.yaml:11]: 5 should be a good compromise. This was done in an attempt to match `rio's` base constraints.

Comment (mbj) [reply core/package.yaml:11]: @epicallan I'll delegate to your experience given RIO uses this also.

File: core/src/RIO/Core.hs (hunk 6 above)
Comment (epicallan) [reply core/src/RIO/Core.hs]: Yeah, going to carry out this change in the colog package as well

Comment (epicallan) [reply core/src/RIO/Core.hs]: Given we are nolonger clashing with `rio` having `MRIO` instead of `MRIO.Core` seems like a good move. What do you think

**Review by mbj — APPROVED** [review]

**Pull request merged**